### DISCUSSION TITLE
Missing The option to set Emulation in "Configure Other" Tab

### DISCUSSION THREAD

**Discussion by NsinghP** [arendst/Tasmota#18267] [Q&A]
Hello, I used Tasmocompiler to compile custom .bin file as I needed to use MQTT over TLS. Everything is good and custom bin file was flashed successfully. However, now I do not see the option to set Emulation in "Configure Other" Tab. I need to set "Hue Emulation Multi device" for Alexa to control these devices. How do I get the Emulation setting back in "configure other Tab"?

**Comment by NsinghP** [comment]
EDIT: Spoke too soon, Solution below doesn't work. It give the option to select emulation but it does not allow to save the selection and bounces back to NONE.

Never mind, figured it out>>>> adding custom parameters seems to have done the trick. Following are the parameters I added to custom .bin file: 

#define MODULE            USER_MODULE
#define EMULATION         EMUL_HUE
#define USE_EMULATION_HUE

Though, I'm not sure if I needed to use the " #define MODULE            USER_MODULE" parameter, but I did it anyways.

**Reply by sfromis** [reply]
`#define MODULE USER_MODULE` is for when you have a `USER_TEMPLATE` to activate.

**Reply by NsinghP** [reply]
> `#define MODULE USER_MODULE` is for when you have a `USER_TEMPLATE` to activate.

Thanks for pointing that out. Seems like the new .bin file that was generated does not want to accept old configuration file. Looks like I will have to enter all the config info manually, what a pain. It may have something to do with the .bin file footprint of 4M versus smaller footprint of previous firmware. This may also be the reason why it keep bouncing to "None" for emulation even though I select "Hue Emulation Multi device" repeatedly. Off to painstakingly entering config info for 3 more devices :(  
PS. I wish it was possible to have a stock version of tasmota firmware for download with MQTT-TLS feature included!

**Reply by sfromis** [reply]
With words like "seems" and "does not want", there is no info for others to guess about what's happening.

**Reply by NsinghP** [reply]
> With words like "seems" and "does not want", there is no info for others to guess about what's happening.

Sorry, With the new custom firmware I get the option to select emulation, but when I select either the "Belkin" Or "Hue Multi Device" it has no effect and just bounces back to "None" after saving is done. The only way to save emulation setting is to reset configuration and then enter everything manually. I'd sure love a way around having to enter everything again. (Though, I have already completed 2 devices the manual way.)
